src/mon/OSDMonitor.cc

```diff
@@ -7784,22 +7784,30 @@ int OSDMonitor::get_crush_rule(const string &rule_name,
   return 0;
 }
 
-int OSDMonitor::check_pg_num(int64_t pool, int pg_num, int size, ostream *ss)
+int OSDMonitor::check_pg_num(int64_t pool, int pg_num, int size, int crush_rule, ostream *ss)
 {
   auto max_pgs_per_osd = g_conf().get_val<uint64_t>("mon_max_pg_per_osd");
-  auto num_osds = std::max(osdmap.get_num_in_osds(), 3u);   // assume min cluster size 3
-  auto max_pgs = max_pgs_per_osd * num_osds;
   uint64_t projected = 0;
-  if (pool < 0) {
-    projected += pg_num * size;
-  }
-  for (const auto& i : osdmap.get_pools()) {
-    if (i.first == pool) {
-      projected += pg_num * size;
-    } else {
-      projected += i.second.get_pg_num_target() * i.second.get_size();
+  unsigned osd_num = 0;
+  auto num_osds = std::max(osdmap.get_num_in_osds(), 3u);
+  CrushWrapper newcrush;
+  if (mapping.get_epoch() >= osdmap.get_epoch()) {
```

Comment:
IIUC, if `mapping` 's epoch is not new enough, `projected` is always zero. i don't think this is expected.

Reply:
This is a mistake, if mapping 's epoch is not new enough, the value of projected will be pg_num * size.